File: app/src/main/java/com/app/airmenu/utils/decryption/APIClient.java
package com.app.airmenu.utils.decryption;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class APIClient {

    static APIInterface INSTANCE = null;

    public static APIInterface getApiInstance() {
        if (INSTANCE == null) {
            INSTANCE = APIClient.getClient().create(APIInterface.class);
        }
        return INSTANCE;
    }

    private static Retrofit getClient() {
        HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor();
        loggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        OkHttpClient client = new OkHttpClient.Builder()
                .readTimeout(60, TimeUnit.SECONDS)
                .addInterceptor(loggingInterceptor)
                .build();

        return new Retrofit.Builder()
                .baseUrl("https://us-central1-airmenu-connector.cloudfunctions.net/")
                .addConverterFactory(GsonConverterFactory.create())
                .client(client)
                .build();
    }

}

File: app/src/main/java/com/app/airmenu/network/APIClient.kt
package com.imprint.app.network

import com.app.airmenu.BuildConfig
import com.google.gson.GsonBuilder
import okhttp3.Interceptor
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import okhttp3.logging.HttpLoggingInterceptor
import java.util.concurrent.TimeUnit

object ApiClient {

    var token: String = ""
    var okHttpClient = OkHttpClient.Builder().apply {
        addInterceptor(
            Interceptor { chain ->
                val builder = chain.request().newBuilder()
                builder.header("Authorization", token)
                    .header("Content-Type", "application/json")
                   // .header("X-Requested-With", "XMLHttpRequest")
                return@Interceptor chain.proceed(builder.build())
            }
        )

        addInterceptor(HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY))
        connectTimeout(55, TimeUnit.SECONDS)
        readTimeout(55, TimeUnit.SECONDS)
        writeTimeout(55,TimeUnit.SECONDS)

    }.build()

    private fun retrofit(): Retrofit {
        return Retrofit.Builder()
            .addConverterFactory(GsonConverterFactory.create(GsonBuilder().create()))
            .baseUrl(BuildConfig.BASE_URL)
            .client(okHttpClient)
            .build()
    }

    val retrofitService: APIInterface by lazy {
        retrofit().create(APIInterface::class.java)
    }

}